openedx/core/djangoapps/content/block_structure/signals.py

```diff
@@ -13,7 +13,7 @@ def _listen_for_course_publish(sender, course_key, **kwargs):  # pylint: disable
     Catches the signal that a course has been published in the module
     store and creates/updates the corresponding cache entry.
     """
-    update_course_in_cache.apply_async([unicode(course_key)], countdown=0)
+    update_course_in_cache.apply_async([unicode(course_key)], {'desired_queue_env': 'lms'}, countdown=0)
```

Comment:
I had to specify `desired_queue_env` here, as task decorators are not available in `route_for_task`. It makes sense though - when calling the task, the calling code has the context to know it needs a queue with settings different than its own.